python_modules/libraries/dagster-sdf/dagster_sdf/information_schema.py

```diff
@@ -0,0 +1,207 @@
+import os
+from pathlib import Path
+from typing import Dict, List, Mapping, Optional, Set, Tuple, Union
+
+import pandas as pd
+from dagster import AssetKey, AssetOut, Nothing, TableColumn, TableSchema
+from dagster._core.definitions.metadata import (
+    CodeReferencesMetadataSet,
+    CodeReferencesMetadataValue,
+    LocalFileCodeReference,
+    TableColumnConstraints,
+    TableMetadataSet,
+)
+from dagster._core.definitions.tags import StorageKindTagSet
+from pydantic import BaseModel, DirectoryPath
+
+from .asset_utils import dagster_name_fn, default_asset_key_fn
+from .constants import (
+    DEFAULT_SDF_WORKSPACE_ENVIRONMENT,
+    SDF_INFORMATION_SCHEMA_TABLES,
+    SDF_TARGET_DIR,
+)
+from .errors import DagsterSdfInformationSchemaNotFoundError
+
+
+class LazyDataFrameDict:
+    def __init__(
+        self,
+        information_schema_dir: Union[Path, str],
+    ):
+        if isinstance(information_schema_dir, str):
+            information_schema_dir = Path(information_schema_dir)
+        self.information_schema_dir = information_schema_dir
+        self.data = {}
+        self.loaded = {}
+
+    def add(self, key: str, dataframe: pd.DataFrame):
+        self.data[key] = dataframe
+        self.loaded[key] = True
+
+    def get(self, key: str) -> pd.DataFrame:
+        if key not in SDF_INFORMATION_SCHEMA_TABLES:
+            raise ValueError(f"Table {key} not valid information schema table.")
+        if key not in self.loaded:
+            self.data[key] = self._load_dataframe(key)
+            self.loaded[key] = True
+        return self.data[key]
+
+    def _load_dataframe(self, key: str) -> pd.DataFrame:
+        # Load the parquet file from disk
+        df_list = [
+            pd.read_parquet(f, engine="pyarrow")
+            for f in self.information_schema_dir.joinpath(key).glob("*.parquet")
+        ]
+        return pd.concat(df_list, ignore_index=True)
+
+    def to_dict(self):
+        # Convert to a serializable dictionary format
+        return {k: v.to_dict() for k, v in self.data.items() if self.loaded[k]}
+
+    @classmethod
+    def from_dict(cls, dict_data):
+        instance = cls()
+        for key, data in dict_data.items():
+            instance.data[key] = pd.DataFrame(data)
+            instance.loaded[key] = True
+        return instance
+
+
+class SdfInformationSchema(BaseModel):
```

Comment:
Docstring please, along with any information about the format of the underlying information schema (e.g. what's available for access, what tables are available, etc)